### ISSUE TITLE
Invites

### ISSUE THREAD

**Issue by ericgio** [yekrojam/api#46]
Add invites to the schema. Edge between user and org like membership. Or could memberships have a pending state?

**Comment by makinde** [comment]
Yeah, good question/topic. I could imagine a couple ways of doing this.

1) Use another addition in `Membership.ROLES`. It could be `INVITED` for instance. Pro: super simple. Cons: Sorta hacky. Also, it means that all code really needs to be on point at check the role of a membership (since presumably an invitee has limited access?).

2) Have an invite just make someone a member right away. Pro: Simple. Con: Could be abused by an admin just adding tons of folks. Maybe add an "accepted" field to the membership?

3) Create a new model that represents the invites. Pro: Clean separation. Con: I sorta like having one places to go and see "which users are affiliated with this org"

4) Prob lots of other options.

**Comment by ericgio** [comment]
From a product perspective, I see admins adding a bunch of people to the org, whether or not those people actually join/use the tool. So the membership would be created, which would then trigger an invitation. A pending state on the membership makes sense to me (like a gray account), but totally open to having a separate model.

**Comment by makinde** [comment]
> which would then trigger an invitation

Product question. What's in that "invitation"? Is it asking the new user to take some action? Or is it more of a notification just telling them that they are in this org now? Where would the pending state be used? I'm sorta wondering if it needs to be pending at all. If they can just be a full member right away, from the time they are invited. There very well might be a workflow I'm not thinking of.

**Comment by ericgio** [comment]
Yeah you may be right that a pending state isn't even needed and this is mainly just a notification to the user that they've been added, though there could be an email confirmation aspect to it as a way to validate the data.

**Comment by makinde** [comment]
That sounds like a good plan. For now, just create new members. If we went people to be able to confirm their membership, we can add a Boolean flag to the model that is `confirmed` or something. It could even be `confirmationDate` if we wanted.

**Comment by ericgio** [comment]
So then the API will need a postCreate hook or something to schedule the email job.

**Comment by makinde** [comment]
Yeah. Prob something like `membershipHooks.js` that schedules agenda jobs.

What do you think of adding a `creator` field to `membership` so that one can see who added the user to the org?

**Comment by ericgio** [comment]
> What do you think of adding a `creator` field to `membership`

Sounds good
